File: hackerrank/python/interview-preparation-kit/string-manipulation/making_anagrams.py
from collections import Counter
import os

# Complete the makeAnagram function below.


def makeAnagram(a, b):
    char_list = list(set(a + b))
    counter_a = Counter(a)
    counter_b = Counter(b)
    count = 0
    for char in char_list:
        count += abs(counter_a[char] - counter_b[char])
    return count


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    a = input()

    b = input()

    res = makeAnagram(a, b)

    fptr.write(str(res) + '\n')

    fptr.close()

File: hackerrank/java/interview-preparation-kit/StringManipulation/MakingAnagrams.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class MakingAnagrams {

    static int makeAnagram(String a, String b) {
        Map<Character, Integer> counterA = new HashMap<>();
        Map<Character, Integer> counterB = new HashMap<>();
        int deletionsCount = 0;

        for(int i = 'a'; i <= 'z'; i++) {
            counterA.put((char) i, 0);
            counterB.put((char) i, 0);
        }

        for(char c : a.toCharArray())
            counterA.put(c, counterA.get(c) + 1);
        
        for(char c : b.toCharArray())
            counterB.put(c, counterB.get(c) + 1);

        for (char letter : counterA.keySet()) {
            int countA = counterA.get(letter);
            int countB = counterB.get(letter);

            deletionsCount += Math.abs(countA - countB);
        }


        return deletionsCount;
    }

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) throws IOException {
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        String a = scanner.nextLine();

        String b = scanner.nextLine();

        int res = makeAnagram(a, b);

        bufferedWriter.write(String.valueOf(res));
        bufferedWriter.newLine();

        bufferedWriter.close();

        scanner.close();
    }
}
